lib/io/hardware_i2c.cpp

```diff
@@ -17,7 +17,7 @@ std::optional<HardwareI2c> HardwareI2c::create(core::ILogger &logger,
                                                const std::uint8_t bus_address)
 {
   char path[13];  // up to "/dev/i2c-2"
-  sprintf(path, "/dev/i2c-%d", bus_address);
+  snprintf(path, sizeof(path), "/dev/i2c-%d", bus_address);
```

Comment:
correct me if i'm wrong, but isn't `sizeof(path)` just the size of the pointer? or has this got to do with the C-array not having decayed just yet?

Reply:
yep its exactly as you suspect